### PR TITLE
include/endian:add openbsd style macro

### PR DESCRIPTION
**Pull request by anjiahao1**
add this macro bemtoh htobem lemtoh htolem

Signed-off-by: anjiahao <anjiahao@xiaomi.com>

## Summary
add some endian macro
## Impact
nothing
## Testing
ci

### CODE REVIEW

File: include/endian.h [hunk 1]
```diff
@@ -140,18 +140,47 @@
...
+#define __bemtoh32(x)       htobe32(*(uint32_t *)(x))
+#define __htobem32(x, v)    (*(uint32_t *)(x) = htobe32(v))
+#define __lemtoh32(x)       htole32(*(uint32_t *)(x))
+#define __htolem32(x, v)    (*(__uint32_t *)(x) = htole32(v))
+
+#ifdef CONFIG_HAVE_LONG_LONG
+#define __bemtoh64(x)       htobe64(*(uint64_t *)(x))
+#define __htobem64(x, v)    (*(uint64_t *)(x) = htobe64(v))
+#define __lemtoh64(x)       htole64(*(uint64_t *)(x))
+#define __htolem64(x, v)    (*(__uint64_t *)(x) = htole64(v))
```
Comment (yamt) [inline include/endian.h:157]: why do you use uint64_t in some places and __uint64_t in the other places?

File: include/endian.h [hunk 2]
```diff
@@ -140,18 +140,47 @@
...
 
 #define betoh16               be16toh
 #define letoh16               le16toh
+#define bemtoh16              __bemtoh16
+#define htobem16              __htobem16
+#define lemtoh16              __lemtoh16
+#define htolem16              __htolem16
 #define betoh32               be32toh
 #define letoh32               le32toh
+#define bemtoh32              __bemtoh32
```
Comment (yamt) [inline include/endian.h:172]: why do you use __bemtoh32 indirection?

**Comment by yamt** [comment]
do you have an example of apps which uses these api?

File: include/endian.h [hunk 3]
```diff
@@ -146,12 +146,24 @@
 
 #define betoh16               be16toh
 #define letoh16               le16toh
+#define bemtoh16(x)           htobe16(*(uint16_t *)(x))
```
Comment (pkarashchenko) [inline include/endian.h:149]: Why this is not depending on big or little endian?

**Comment by anjiahao1** [comment]
> do you have an example of apps which uses these api?

[crypto](https://github.com/apache/incubator-nuttx/actions/runs/3343702821/jobs/5537455268) crypto use it

File: include/endian.h (hunk 3 above)
Comment (anjiahao1) [reply include/endian.h:149]: Basic macros are already defined

Comment (pkarashchenko) [reply include/endian.h:149]: Sorry. Missed that

File: include/endian.h [hunk 4]
```diff
@@ -146,12 +146,24 @@
...
+#define htobem16(x, v)        (*(uint16_t *)(x) = htobe16(v))
+#define lemtoh16(x)           htole16(*(uint16_t *)(x))
```
Comment (yamt) [inline include/endian.h:151]: i guess it's simpler to read if you use letoh16 instead of htole16.
ditto for other macros.

**Comment by yamt** [comment]
> > do you have an example of apps which uses these api?
> 
> [crypto](https://github.com/apache/incubator-nuttx/actions/runs/3343702821/jobs/5537455268) crypto use it

ok. thank you

File: include/endian.h (hunk 3 above)
Comment (xiaoxiang781216) [inline include/endian.h:149]: FAR

**Review by xiaoxiang781216 — APPROVED** [review]

**Review by jerpelea — APPROVED** [review]

File: include/endian.h [hunk 5]
```diff
@@ -146,12 +146,24 @@
 
 #define betoh16               be16toh
 #define letoh16               le16toh
+#define bemtoh16(x)           htobe16(*(FAR uint16_t *)(x))
```
Comment (pkarashchenko) [inline include/endian.h:149]: Whyit is not

```suggestion
#define bemtoh16(x)           betoh16(*(FAR uint16_t *)(x))
```
Even it it results to the same it seems to be not intuitive as it is now

File: include/endian.h [hunk 6]
```diff
@@ -146,12 +146,24 @@
...
+#define htobem16(x, v)        (*(FAR uint16_t *)(x) = htobe16(v))
+#define lemtoh16(x)           htole16(*(FAR uint16_t *)(x))
```
Comment (pkarashchenko) [inline include/endian.h:151]: Why not
```suggestion
#define lemtoh16(x)           letoh16(*(FAR uint16_t *)(x))
```

**Review by pkarashchenko — COMMENTED** [review]
same comment is applied to all other similar places in this PR

**Review by xiaoxiang781216 — APPROVED** [review]

**Pull request merged**